app/charts/map/map.tsx

```diff
@@ -116,6 +117,38 @@ export const MapComponent = () => {
     }
   }, []);
 
+  const zoomToShapes = useCallback(() => {
+    if (features.areaLayer?.shapes) {
+      const bbox = geoBounds(features.areaLayer.shapes);
+
+      setViewState(constrainZoom(viewState, bbox, { padding: 48 }));
+    }
+  }, [viewState, features.areaLayer?.shapes]);
+
+  const zoomToSymbols = useCallback(() => {
+    if (features.symbolLayer?.points) {
+      const visiblePoints = features.symbolLayer.points.filter(
+        (d) => d.properties.observation !== undefined
+      );
+      const [minLng, maxLng] = extent(visiblePoints, (d) => d.coordinates[0]);
+      const [minLat, maxLat] = extent(visiblePoints, (d) => d.coordinates[1]);
+      const bbox = [
+        [minLng, minLat],
+        [maxLng, maxLat],
+      ] as [[number, number], [number, number]];
+
+      setViewState(constrainZoom(viewState, bbox, { padding: 48 }));
+    }
+  }, [viewState, features.symbolLayer?.points]);
+
+  const initialZoom = useCallback(() => {
+    if (areaLayer.show) {
+      zoomToShapes();
+    } else if (symbolLayer.show) {
+      zoomToSymbols();
+    }
+  }, [areaLayer.show, symbolLayer.show, zoomToShapes, zoomToSymbols]);
+
```

Comment:
I am not proud of this solution, as this function only gets called once, when DeckGL instance is loaded; but there was a need to access `ViewState`, as the first event that is fired is `onResize` event, which updates the state so we can't use `INITIAL_VIEW_STATE`...

`onLoad` doesn't provide an access to viewState either – let me know if you have an idea to improve 🙏

Reply:
Both area layers and symbol layers can be shown right ? So in this case, we should include in the bounding box both types right ? Maybe onLoad can be used just to track loading status of the map and then zoom can be managed in an effect ? To ensure calling only once, we can use a ref.

```
const hasZoomedRef = useRef()
const setInitialZoom = useCallback(() => {
  if (hasZoomedRef.current) { return }
  const bb = makeBoundingBox(showSymbolLayer ? symbolLayer : undefined, showAreaLayer ? areaLayer : undefined)
 zoomToBoundingBox(bb)
 hasZoomedRef.current = true
}, [mapLayer, symbolLayer, areaLayer])

useEffect(() => {
  if (!mapLoaded || !symbolLayerLoaded || !areaLayerLoaded) {
    return
  }
  setInitialZoom()
}, [mapLoaded, symbolLayerLoaded, areaLayerLoaded])

...
<StaticMap
  onLoad={() => setMapLoaded(true)}
/>
```

Reply:
Ah I forgot view state in my code but it could easily be integrated in the effect.